pyccel/ast/builtin_methods/list_methods.py

```diff
@@ -306,3 +307,44 @@ def __init__(self, list_obj) -> None:
         self._order = list_obj.order
         self._class_type = list_obj.class_type
         super().__init__(list_obj)
+
+#==============================================================================
+class ListSort(ListMethod) :
+    """
+    Represents a call to the .sort() method.
+
+    Represents a call to the `.sort()` method, which sorts the elements of the
+    list in ascending order and modifies the original list in place. This means
+    that the elements of the original list are rearranged to be in sorted order.
+    Optional parameters are not supported, therefore they should not be provided. 
+    Note that the .sort() method doesn't return any value.
+    
+    >>> a = [3, 1, 4, 1, 5, 9, 2, 6, 5, 3, 5]
+    >>> a.sort()
+    >>> print(a)
+    [1, 1, 2, 3, 3, 4, 5, 5, 5, 6, 9]
+
+    Parameters
+    ----------
+    list_obj : TypedAstNode
+        The list object which the method is called from.
+
+    reverse : TypedAstNode, optional
+        Argument mimicking sort's reverse parameter. This argument is 
+        unsupported so it should not be provided.
+
+    key : FunctionDef, optional
+        A function to specify the sorting criteria(s). This argument is 
+        unsupported so it should not be provided.
+    """
+    __slots__ = ()
+    _rank = 0
+    _order = None
+    _shape = None
+    _class_type = VoidType()
+    name = 'sort'
```

Comment:
Beware. This code needs changing now that #1808 is merged

Reply:
updated 
792d7f449b75eedbcf0170c195307badc732de8f